src/System.Windows.Extensions/src/System/Drawing/FontConverter.cs

```diff
@@ -126,94 +123,179 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
+                    styleStr = font.Substring(styleIndex, font.Length - styleIndex);
+
+                    // Expected style format ~ "style=Italic, Bold"
+                    if (!styleStr.StartsWith(StyleHdr, StringComparison.CurrentCultureIgnoreCase))
                     {
-                        f_size = (float)TypeDescriptor.GetConverter(typeof(float)).ConvertFromString(context, culture, fields[1].Substring(0, i));
-                        units = fields[1].Substring(i);
-                        break;
+                        throw new ArgumentException(SR.Format(SR.TextParseFailedFormat, font, $"name{separator} size[units[{separator} style=style1[{separator} style2{separator} ...]]]"), nameof(styleStr));
                     }
+
+                    // Get the mid-substring containing the size information.
+                    sizeStr = font.Substring(nameIndex + 1, styleIndex - nameIndex - 1);
                 }
-                switch (units)
+                else
                 {
-                    case "display":
-                        f_unit = GraphicsUnit.Display;
-                        break;
+                    // no style.
+                    sizeStr = font.Substring(nameIndex + 1);
+                }
 
-                    case "doc":
-                        f_unit = GraphicsUnit.Document;
-                        break;
+                // Parse size.
+                (string size, string unit) unitTokens = ParseSizeTokens(sizeStr, separator);
 
-                    case "pt":
-                        f_unit = GraphicsUnit.Point;
-                        break;
+                if (unitTokens.size!= null)
+                {
+                    try
+                    {
+                        fontSize = (float)TypeDescriptor.GetConverter(typeof(float)).ConvertFromString(context, culture, unitTokens.size);
+                    }
+                    catch
+                    {
+                        // Exception from converter is too generic.
+                        throw new ArgumentException(SR.Format(SR.TextParseFailedFormat, font, $"name{separator} size[units[{separator} style=style1[{separator} style2{separator} ...]]]"), nameof(sizeStr));
+                    }
+                }
 
-                    case "in":
-                        f_unit = GraphicsUnit.Inch;
-                        break;
+                if (unitTokens.unit != null)
+                {
+                    // ParseGraphicsUnits throws an ArgumentException if format is invalid.
+                    units = ParseGraphicsUnits(unitTokens.unit);
+                }
 
-                    case "mm":
-                        f_unit = GraphicsUnit.Millimeter;
-                        break;
+                if (styleStr != null)
+                {
+                    // Parse FontStyle                        
+                    int eqIndex = styleStr.IndexOf("=");
+                    styleStr = styleStr.Substring(eqIndex + 1);
```

Comment:
this previously did not assume that this was the end of the string: `styleStr = styleStr.Substring(eqIndex + 1, styleStr.Length - styleHdr.Length);` https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/FontConverter.cs,151
intentional?

Reply:
Actually its the same thing as ```eqIndex + 1``` will always be equal to stylHdr.Length